lib/db/set.go

```diff
@@ -380,6 +381,44 @@ func (s *Snapshot) RemoteNeedFolderFiles(device protocol.DeviceID, page, perpage
 	return files
 }
 
+func (s *Snapshot) WithBlocksHash(hash []byte, iterator Iterator) error {
+	key, err := s.t.keyer.GenerateBlockListMapKey(nil, []byte(s.folder), hash, nil)
```

Comment:
It would be more consistent with all the other iterators if the implementation lived in `transactions.go` and has the usual ```Debugf(`%s WithBlocksHash ...``` logging here.

Reply:
There is nothing of the snapshot package in that file:

```
/cygdrive/c/Gohome/src/github.com/syncthing/syncthing (renames) $ cat lib/db/transactions.go | grep Snapshot | wc -l
0
```

Seems like not the right place to land that?

Reply:
I guess you suggest to split it out. Right, let me do that.